state/executor.go

```diff
@@ -531,6 +543,11 @@ func (t *Transition) Call2(
 }
 
 func (t *Transition) run(contract *runtime.Contract, host runtime.Host) *runtime.ExecutionResult {
+	// check allow list (if any)
+	if t.deploymentAllowlist != nil && t.deploymentAllowlist.Addr() == contract.CodeAddress {
```

Comment:
This enables the allow list to be reachable from other smart contracts (pending) and from transactions made by the user.